components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/RemoteFetchConfigurationServiceImpl.java

```diff
@@ -138,8 +139,25 @@ public RemoteFetchConfiguration getRemoteFetchConfiguration(String fetchConfigur
     public void deleteRemoteFetchConfiguration(String fetchConfigurationId)
             throws RemoteFetchCoreException {
 
+        RemoteFetchTaskExecutor.getInstance().deleteRemoteFetchConfigurationFromBatchTask(fetchConfigurationId);
+
         int tenantId = IdentityTenantUtil.getTenantId(CarbonContext.getThreadLocalCarbonContext().getTenantDomain());
         this.fetchConfigurationDAO.deleteRemoteFetchConfiguration(fetchConfigurationId, tenantId);
     }
 
+    /**
+     * This method used to get remote fetch configuration for given id and start an Immediate task execution.
+     * @param fetchConfigurationId
+     * @throws RemoteFetchCoreException
+     */
+    @Override
+    public void triggerRemoteFetchConfiguration(String fetchConfigurationId) throws RemoteFetchCoreException {
+
+        RemoteFetchConfiguration remoteFetchConfiguration = this.getRemoteFetchConfiguration(fetchConfigurationId);
+        if (remoteFetchConfiguration != null) {
+            RemoteFetchTaskExecutor.getInstance().startImmediateTaskExecution(remoteFetchConfiguration);
+        }
```

Comment:
We need a debug log to see what happens.

Reply:
Debug log added.

Reply:
How to see there was no RemoteFetchConfiguration for the given id?